### PR TITLE
Feat(CLI) Icon Check fix: #309

### PR DESCRIPTION
**Pull request by tensor-programming**
**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [x] Yes. Issue #309
- [ ] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

File: cli/tauri.js/templates/src-tauri/src/build.rs [hunk 1]
```diff
@@ -3,10 +3,13 @@ extern crate winres;
 
 #[cfg(windows)]
 fn main() {
-  let mut res = winres::WindowsResource::new();
-  res.set_icon("icons/icon.ico");
-  res.compile().expect("Unable to find visual studio tools");
-}
+  if std::path::Path::new("icons/icon.ico").exists() {
```
Comment (nothingismagick) [inline cli/tauri.js/templates/src-tauri/src/build.rs]: This hardcodes the `icons/icon.ico` location. We are specifying its location in the `src-tauri/Cargo.toml` file.

Comment (tensor-programming) [reply cli/tauri.js/templates/src-tauri/src/build.rs]: I know, I am working on that.

Comment (nothingismagick) [reply cli/tauri.js/templates/src-tauri/src/build.rs]: hmm. wouldn't it be better to do that in this loop:
`cli/tauri-cli/src/bundle/settings.rs`
```
  /// Returns an iterator over the icon files to be used for this bundle.
  pub fn icon_files(&self) -> ResourcePaths<'_> {
    match self.bundle_settings.icon {
      Some(ref paths) => ResourcePaths::new(paths.as_slice(), false),
      None => ResourcePaths::new(&[], false),
    }
  }
```

**Comment by lucasfernog** [comment]
Are you working on the tauri-bundler check too?

File: cli/tauri.js/templates/src-tauri/src/build.rs (hunk 1 above)
Comment (tensor-programming) [reply cli/tauri.js/templates/src-tauri/src/build.rs]: Not that function but yes, in the bundler is where this will go.

**Comment by tensor-programming** [comment]
Yes, this will be implemented on the bundler.  let me know if there are any other pieces you want me to add to the bundler to stop the compilation.

**Comment by lucasfernog** [comment]
The only thing that I remember is something from the external binary bundling, which is not ready yet. So we're ok with this PR.

**Pull request merged**